controllers/vrg_volrep_test.go

```diff
@@ -1073,6 +1073,15 @@ func (v *vrgTest) createPVC(pvcName, namespace, volumeName string, labels map[st
 	Expect(err).To(SatisfyAny(BeNil(), MatchError(expectedErr)),
 		"failed to create PVC %s", pvcName)
 
+	// Make sure the PVC is created to avoid any timing issues
+	Eventually(func() error {
```

Comment:
We just created the PVC, it should either error out in the Create or the Get should not have to wait. This seems incorrect as the fix for the issue reported.

If we chase the logs from the issue at https://github.com/RamenDR/ramen/actions/runs/5706933517/job/15463123972?pr=1007 we see that the message for `STEP: creating PVC pv-xxoyu-03` is not present. This seems to indicate the 4th PVC with the suffix 03 was not created at all, justifying the failure in Get.

Looking at the code in `VRGTestCaseStart` I cannot find any reason why the PVC with index 03 was not created, but the failure seems to stem from there, and not from waiting for a Get to succeed.

Reply:
@ShyamsundarR , according to logs, you mentioned above, there was a creation -   STEP: creating PVC pvc-xxoyu-03 @ 07/30/23 15:15:23.337 (line 181). And the failure is here: [FAILED] in [It] - /home/runner/work/ramen/ramen/controllers/vrg_volrep_test.go:1274 @ 07/30/23 15:15:23.341, it's 4 ms after the creation. Maybe, it takes more time to create?

Reply:
True... I searched the logs a few times and did not find the above. Maybe I was looking in the wrong place. I do see that the PVC/PV pair for all 4 volumes (00-03) are created.

Create call if it returns, it signals a successful create, there is no delayed creation in this case. The issue though is which [client](https://sdk.operatorframework.io/docs/building-operators/golang/references/client/#default-client) is used to fetch the PVC/PV or any object for that matter. If it is reading from the cache then it would take time to succeed based on when the cache was updated. So the issue here is that something should retry the operation on such failures than consider it a final failure. Will look at the code a bit more.

Reply:
@ShyamsundarR Done!